app/src/main/java/com/example/sharewhatyoucanproject/LocationTrack.kt

```diff
@@ -0,0 +1,250 @@
+import android.app.Service
+import android.graphics.Bitmap
+import android.provider.MediaStore
+import android.content.ContentResolver
+import android.content.Context
+import android.location.Location
+import android.webkit.MimeTypeMap
+import com.google.android.gms.tasks.OnSuccessListener
+import com.example.sharewhatyoucanproject.ImageUploadInfo
+import com.google.android.gms.tasks.OnFailureListener
+import com.google.firebase.storage.OnProgressListener
+import android.location.LocationListener
+import android.location.LocationManager
+import androidx.core.app.ActivityCompat
+import android.os.IBinder
+import android.provider.Settings
+import com.example.sharewhatyoucanproject.DashboardActivity
+import java.lang.Exception
+
+class LocationTrack(private val mContext: Context) : Service(), LocationListener {
+    var checkGPS = false
+    var checkNetwork = false
+    var canGetLocation = false
+    var loc: Location? = null
+    private var latitude = 0.0
+    private var longitude = 0.0
+    protected var locationManager: LocationManager? = null// TODO: Consider calling
+    //    ActivityCompat#requestPermissions
+    // here to request the missing permissions, and then overriding
+    //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+    //                                          int[] grantResults)
+    // to handle the case where the user grants the permission. See the documentation
+    // for ActivityCompat#requestPermissions for more details.
+
+
+    /*if (checkNetwork) {
+
+
+           if (ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+               // TODO: Consider calling
+               //    ActivityCompat#requestPermissions
+               // here to request the missing permissions, and then overriding
+               //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+               //                                          int[] grantResults)
+               // to handle the case where the user grants the permission. See the documentation
+               // for ActivityCompat#requestPermissions for more details.
+           }
+           locationManager.requestLocationUpdates(
+                   LocationManager.NETWORK_PROVIDER,
+                   MIN_TIME_BW_UPDATES,
+                   MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
+
+           if (locationManager != null) {
+               loc = locationManager
+                       .getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+
+           }
+
+           if (loc != null) {
+               latitude = loc.getLatitude();
+               longitude = loc.getLongitude();
+           }
+       }*/
+// if GPS Enabled get lat/long using GPS Services
+    // get GPS status
+
```

Comment:
Why a code cemetery here?  If you really want to possibly uncomment-out this code later, then instead create a jira ticket and put this code snippet in the ticket.